solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1735,7 +1736,6 @@ static boolean sendToOverseer(DocCollection coll, String replicaName) {
     if (r == null) return true;
     Slice shard = coll.getSlice(r.slice);
     if (shard == null) return true;//very unlikely
-    if (shard.getState() == Slice.State.RECOVERY) return true;
```

Comment:
we don't need this now?

Reply:
I don't think so, removing this and letting individual nodes update PRS state seems to be good enough, it doesn't seem like sending to overseer during recovery actually changes anything. I've tested a number of operations and not seeing any difference but still a few more tests to run.

Reply:
this was added as a defensive check when shard split is going on. This is actually not required